### PR TITLE
fix: Do not add trailing slash in xrootd urls

### LINKED ISSUE #1053: Too many slashes in file urls from dataset_tools
**Describe the bug**
Using the new dataset query API through Rucio returns PFNs with one too many slashes, e.g.
```
│   ├── root://xrootd.cmsaf.mit.edu:1094///store/data/Run2018D/SingleMuon/NANOAOD/UL2018_MiniAODv2_NanoAODv9_GT36-v1/70000/5E570C94-EF7F-F64E-851B-AE9074B89A30.root
```
For some sites, such as MIT, this can cause strange opening/reading errors. Generally xrootd urls should have one or two slashes between the hostid and the path, e.g.
```
root://xrootd.cmsaf.mit.edu:1094//store/data/Run2018D/SingleMuon/NANOAOD/UL2018_MiniAODv2_NanoAODv9_GT36-v1/70000/5E570C94-EF7F-F64E-851B-AE9074B89A30.root
```
This may be site-specific, and perhaps is an issue on the Rucio client side.

### PR DESCRIPTION
**Pull request by valsdav**
Now the sites map cache file is invalidated after 10 mins to avoid stale information. 

Fixes #1053

### CODE REVIEW

**Comment by lgray** [comment]
@nsmith- can you give this a try?

File: src/coffea/dataset_tools/rucio_utils.py [hunk 1]
```diff
@@ -114,7 +124,8 @@ def _get_pfn_for_site(path, rules):
                     pfn = pfn.replace(f"${i+1}", grs[i])
                 return pfn
     else:
-        return rules + "/" + path
+        # not adding any slash as the path usually starts with it
+        return rules + path
```
Comment (nsmith-) [inline src/coffea/dataset_tools/rucio_utils.py]: To be sure you could use `rules + "/" + path.removeprefix("/")`

**Review by nsmith- — APPROVED** [review]
I should point out that rather than parsing the SITECONF you can also ask rucio to give you the PFNs, e.g. on the commandline
```
rucio list-file-replicas cms:/SingleMuon/Run2018A-UL2018_MiniAODv2_NanoAODv9_GT36-v1/NANOAOD --rse 'rse_type=DISK&country=US' --protocol root
```
gives xrootd urls for files in that dataset

**Comment by valsdav** [comment]
> I should point out that rather than parsing the SITECONF you can also ask rucio to give you the PFNs, e.g. on the commandline
> 
> ```
> rucio list-file-replicas cms:/SingleMuon/Run2018A-UL2018_MiniAODv2_NanoAODv9_GT36-v1/NANOAOD --rse 'rse_type=DISK&country=US' --protocol root
> ```
> 
> gives xrootd urls for files in that dataset

I agree, but the SITCONF hack was done to overcome the issue that not all the sites are returning `root` endpoints, sometimes only `davs`. Maybe the problem has been fixed in the meanwhile?

**Comment by nsmith-** [comment]
> > I should point out that rather than parsing the SITECONF you can also ask rucio to give you the PFNs, e.g. on the commandline
> > ```
> > rucio list-file-replicas cms:/SingleMuon/Run2018A-UL2018_MiniAODv2_NanoAODv9_GT36-v1/NANOAOD --rse 'rse_type=DISK&country=US' --protocol root
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > gives xrootd urls for files in that dataset
> 
> I agree, but the SITCONF hack was done to overcome the issue that not all the sites are returning `root` endpoints, sometimes only `davs`. Maybe the problem has been fixed in the meanwhile?

Is there a ticket/thread with CMS DM ops / site support on the subject? If not, we should make one.

**Comment by valsdav** [comment]
> > > I should point out that rather than parsing the SITECONF you can also ask rucio to give you the PFNs, e.g. on the commandline
> > > ```
> > > rucio list-file-replicas cms:/SingleMuon/Run2018A-UL2018_MiniAODv2_NanoAODv9_GT36-v1/NANOAOD --rse 'rse_type=DISK&country=US' --protocol root
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > gives xrootd urls for files in that dataset
> > 
> > 
> > I agree, but the SITCONF hack was done to overcome the issue that not all the sites are returning `root` endpoints, sometimes only `davs`. Maybe the problem has been fixed in the meanwhile?
> 
> Is there a ticket/thread with CMS DM ops / site support on the subject? If not, we should make one.

I will give it another try, it was more than 1 year ago and I haven't test it since then..

**Pull request merged**